cli/azd/pkg/project/service_target_dotnet_containerapp.go

```diff
@@ -202,7 +202,11 @@ func (at *dotnetContainerAppTarget) Deploy(
 					"urlHost":          fns.UrlHost,
 					"connectionString": fns.ConnectionString,
 					"parameter":        fns.Parameter,
-					"secretOutput":     fns.kvSecret,
+					// secured_parameter gets a parameter the same way as parameter, but supporting the secured_parameter
+					// allows to update the logic of pulling secret parameters in the future, if azd changes the way it
+					// stores the parameter value.
+					"secured_parameter": fns.Parameter,
```

Comment:
Not sure that I love pushing this decision into the template instead of just having `fns.Parameter` "do the right thing" for a secret parameter if we want to change either how we fetch the value or how we give it back to the template.  Would that work?

I guess on some level this may be moot since long term this code will go away when Aspire starts generating ACA definitions themselves?

Reply:
If we do want to keep this around I might spell it as `secureParameter` instead of `secured_parameter`, it feels a little more in line with the others that way...

Reply:
> instead of just having fns.Parameter "do the right thing"

We can keep `parameter only`. All we need to know is if the parameter to fetch is secured or not.  I might be able to set a bool like `func Parameter(name string, secured bool)`. Then we would just write the condition true/false on the template.

This logic won't go away with Aspire generation ACA definitions. If the definition uses a parameter from the main.bicep, azd will still need to create the mappings for the input parameters for the ACA connected to the parameter from main.bicep

Reply:
> All we need to know is if the parameter to fetch is secured or not.

Isn't that a property of the parameter itself?  i.e. Couldn't `func Parameter(name string)` look at the name of the parameter and the manifest determine if it is secure or not?  I'm just wondering why the caller of this function needs to provide this information, isn't it something the callee could determine?

Reply:
the manifest is not always available. For example, after `infra synth` (or for a non-aspire scenario).
So, the parameter function would depend on the infrastructure (bicep), which is not ideal during deployment.  That's why we use the yaml go template as some sort of intermediate state from the infrastructure.

Reply:
Gotcha - do you mind just renaming this to `securedParmater` (camel case) then?

Reply:
Oh yes!  I was forgetting about the change to camelCase. I'll do it